pkg/cvo/upgradeable.go

```diff
@@ -234,6 +247,126 @@ func (check *clusterManifestDeleteInProgressUpgradeable) Check() *configv1.Clust
 	return nil
 }
 
+func gateApplicableToCurrentVersion(gateName string, currentVersion string) (bool, error) {
+	var applicable bool
+	if ackVersion := adminAckGateRegexp.FindString(gateName); ackVersion == "" {
+		return false, fmt.Errorf("%s configmap gate name %s has invalid format; must begin with 'ack-X.Y'",
+			internal.AdminGatesConfigMap, gateName)
+	} else {
+		parts := strings.Split(ackVersion, "-")
+		if strings.HasPrefix(currentVersion, parts[1]) {
+			applicable = true
+		}
+	}
+	return applicable, nil
+}
+
+func checkAdminGate(gate string, currentVersion string,
+	gateConfigmap *corev1.ConfigMap, ackConfigmap *corev1.ConfigMap,
+	ackConfigmapErr error) *configv1.ClusterOperatorStatusCondition {
+
+	if applies, err := gateApplicableToCurrentVersion(gate, currentVersion); err == nil {
+		if !applies {
+			return nil
+		}
+	} else {
+		klog.Error(err)
+		return &configv1.ClusterOperatorStatusCondition{
+			Type:    configv1.ClusterStatusConditionType("UpgradeableAdminAckRequired"),
+			Status:  configv1.ConditionFalse,
+			Reason:  "AdminAckConfigMapIssue",
```

Comment:
`Issue` -> `Error`?  Because we are failing out here, so no need to soft-pedal things.

Reply:
Done